### PR TITLE
Fixes 11973: Pass required parameters while patching the description of a TestCase

### PR DESCRIPTION
**Pull request by gauthk6**
### Describe your changes:

Fixes https://github.com/open-metadata/OpenMetadata/issues/11973
The main fix was to pass the required parameters while patching the description of a TestCase. Also added a test for the same.

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

- [X] I have added a test that covers the exact scenario we are fixing.

### CODE REVIEW

File: docker/development/docker-compose.yml [hunk 1]
```diff
@@ -38,8 +39,7 @@ services:
       timeout: 10s
       retries: 10
     volumes:
-     - ./docker-volume/db-data:/var/lib/mysql
-
+     - db-data:/var/lib/mysql
```
Comment (TeddyCr) [inline docker/development/docker-compose.yml:42]: @gauthk6 we'll need to revert that to keep it as it was. 😊

**Comment by pmbrull** [comment]
Thanks for the patch @gauthk6. I see that the checkstyle is failing though. Please follow the bot instructions to get that cleaned up and we should be good to go

**Comment by gauthk6** [comment]
@TeddyCr I accidentally made the documentation changes in the same branch as the code. I tried to revert it but doesn't seem to work. I will be careful in the future to keep units of work in separate branches and PRs.

**Comment by TeddyCr** [comment]
> @TeddyCr I accidentally made the documentation changes in the same branch as the code. I tried to revert it but doesn't seem to work. I will be careful in the future to keep units of work in separate branches and PRs.

@gauthk6 no worries. You'll just need to fix the conflict and then we can merge it.

File: openmetadata-docs/content/v1.1.0/quick-start/local-deployment.md [hunk 2]
```diff
@@ -145,6 +145,7 @@ pip3 install --upgrade pip setuptools
 ```
 
 ### 5. Install the OpenMetadata Python module using pip
+On macOS, make sure to install XCode prior to running the following command:
```
Comment (pmbrull) [inline openmetadata-docs/content/v1.1.0/quick-start/local-deployment.md:148]: nit - if we ask to install something, either add the installation command or a link to external docs that show how to get this installed. Thanks

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**